src/frontend/Sider/Behandling/RevurderFra/TilsynBarn/useHentFullstendigOversikt.tsx

```diff
@@ -0,0 +1,32 @@
+import { useCallback, useState } from 'react';
+
+import { useApp } from '../../../../context/AppContext';
+import { byggTomRessurs, Ressurs } from '../../../../typer/ressurs';
+import {
+    InnvilgelseBarnetilsyn,
+    VedtakBarnetilsyn,
+} from '../../../../typer/vedtak/vedtakTilsynBarn';
+
+export const useHentFullstendigOversikt = (): {
+    hentBeregningsresultat: (forrigeBehandlingId: string) => void;
+    vedtakBarnetilsyn: Ressurs<VedtakBarnetilsyn>;
+} => {
+    const { request } = useApp();
+
+    const [vedtakBarnetilsyn, setVedtakBarnetilsyn] =
+        useState<Ressurs<VedtakBarnetilsyn>>(byggTomRessurs());
+
+    const hentForrigeVedtakMedFullHistorikk = useCallback(
+        (forrigeBehandlingId: string) => {
+            request<InnvilgelseBarnetilsyn, null>(
+                `/api/sak/vedtak/tilsyn-barn/fullstendig-oversikt/${forrigeBehandlingId}`
+            ).then(setVedtakBarnetilsyn);
+        },
+        [request]
+    );
+
```

Comment:
Kan man kanskje ha useEffect her om at man henter dataen når komponenten initieres? Så trenger man ikke en useEffect i komponenten som bruker denne? 
Og at forrigeBehandlingId blir sendt inn til `useHentFullstendigOversikt`

Reply:
Er dette nødvendig, eller nice to have?